GVFS/GVFS/CommandLine/UpgradeVerb.cs

```diff
@@ -0,0 +1,384 @@
+                    }
+
+                    return false;
+                },
+                "Launching upgrade tool",
+                suppressGvfsLogMessage: true);
+            
+            if (!preUpgradeSuccess)
+            {
+                consoleError = errorMessage;
+                return false;
+            }
+
+            this.ReportInfoToConsole($"{Environment.NewLine}Installer launched in a new window. Do not run any git or gvfs commands until the installer has completed.");
+            consoleError = null;
+            return true;
+        }
+
+        private bool TryCopyUpgradeTool(out string upgraderExePath, out string consoleError)
+        {
+            upgraderExePath = null;
+
+            using (ITracer activity = this.tracer.StartActivity(nameof(this.TryCopyUpgradeTool), EventLevel.Informational))
+            {
+                if (!this.upgrader.TrySetupToolsDirectory(out upgraderExePath, out consoleError))
+                {
+                    return false;
+                }
+
+                activity.RelatedInfo($"Successfully Copied upgrade tool to {upgraderExePath}");
+            }
+
+            return true;
+        }
+
+        private bool TryLaunchUpgradeTool(string path, out string consoleError)
+        {
+            using (ITracer activity = this.tracer.StartActivity(nameof(this.TryLaunchUpgradeTool), EventLevel.Informational))
+            {
+                Exception exception;
+                if (!this.processLauncher.TryStart(path, out exception))
+                {
+                    if (exception != null)
+                    {
+                        consoleError = exception.Message;
+                        this.tracer.RelatedError($"Error launching upgrade tool. {exception.ToString()}");
+                    }
+                    else
+                    {
+                        consoleError = "Error launching upgrade tool";
+                    }
+
+                    return false;
+                }
+
+                activity.RelatedInfo("Successfully launched upgrade tool.");
+            }
+                
+            consoleError = null;
+            return true;
+        }
+
+        private bool TryCheckUpgradeAvailable(
+            out Version latestVersion, 
+            out string consoleError)
+        {
+            latestVersion = null;
+            consoleError = null;
+
+            using (ITracer activity = this.tracer.StartActivity(nameof(this.TryCheckUpgradeAvailable), EventLevel.Informational))
+            {
+                bool checkSucceeded = false;
+                Version version = null;
+
+                checkSucceeded = this.upgrader.TryGetNewerVersion(out version, out consoleError);
+                if (!checkSucceeded)
+                {
+                    return false;
+                }
+
+                latestVersion = version;
+
+                activity.RelatedInfo("Successfully checked server for GVFS upgrades.");

```

Comment:
This log statement should include the version that was discovered.

In general, I'm not sure that the logs we're generating right now are all that useful for debugging user errors. Have you thought through what information needs to be included in them so we can diagnose user issues that get reported later?

Reply:
Added version info.

Also, please see my comment below. This file will contain detailed exception info, which would be missing in the Console logs.